sdk/storage/azdatalake/filesystem/client.go

```diff
@@ -202,7 +203,15 @@ func (fs *Client) BlobURL() string {
 // The new directory.Client uses the same request policy pipeline as the Client.
 func (fs *Client) NewDirectoryClient(directoryPath string) *directory.Client {
 	directoryPath = strings.ReplaceAll(directoryPath, "\\", "/")
-	dirURL := runtime.JoinPaths(fs.generatedFSClientWithDFS().Endpoint(), directoryPath)
+	names := strings.Split(directoryPath, "/")
+	path := ""
+	for i, name := range names {
+		if i > 0 {
+			path += "/"
+		}
+		path += url.PathEscape(name)
```

Comment:
Instead of doing string concat at each step, you can escape all parts and then join the string back with "/" as seperator. Code will look clearner with ath I beleive,

Reply:
made the change